hal/src/rtl872x/adc_hal.cpp

```diff
@@ -293,3 +293,11 @@ int hal_adc_sleep(bool sleep, void* reserved) {
 int hal_adc_calibrate(uint32_t reserved, void* reserved1) {
     return Adc::instance().calibration();
 }
+
+int hal_adc_set_reference(uint32_t reference, void* reserved) {
+    return SYSTEM_ERROR_NOT_SUPPORTED;
+}
+
+int hal_adc_get_reference(void* reserved) {
+    return (int)HAL_ADC_REFERENCE_DEFAULT;
```

Comment:
Probably better to specify which one is the default. I think for rtl872x that is probably internal.